instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/datasource/OpenTelemetryDataSource.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Includes work from:
+/*
+ * Copyright 2017-2021 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.opentelemetry.instrumentation.jdbc.datasource;
+
+import static io.opentelemetry.instrumentation.jdbc.internal.DataSourceSingletons.instrumenter;
+import static io.opentelemetry.instrumentation.jdbc.internal.JdbcUtils.computeDbInfo;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.jdbc.internal.CheckedCallable;
+import io.opentelemetry.instrumentation.jdbc.internal.DbInfo;
+import io.opentelemetry.instrumentation.jdbc.internal.OpenTelemetryConnection;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.logging.Logger;
+import javax.sql.DataSource;
+
+public class OpenTelemetryDataSource implements DataSource, AutoCloseable {
```

Comment:
Can this be package private and final?

Reply:
Oh looks like it's another package, we could probably move it if it'll allow hiding it

Reply:
Actually not, this one is another way to use OpenTelemetry. One way is by using the Driver and another way is to wrap the DataSource, same solution as to how P6Spy works: https://p6spy.readthedocs.io/en/latest/integration.html#datasource-way

I will update the README with covering this use case.

Reply:
Readme updated.